### PR TITLE
BUG: AssignmentSet may not have seen a package_id

### PR DESCRIPTION
**Pull request by johntyree**
The package_ids we consider come from the Clause literals,
but we look them up in the AssignmentSet. If the
AssignmentSet has never seen that particular id before, it
crashes. This fixes that.

### CODE REVIEW

**Comment by cournape** [comment]
I wonder if it makes sense to handle this at the `AssignementSet` level ?

**Comment by johntyree** [comment]
Hmm...

Yes it probably does, given that `AssignmentSet.value` returns `None` for unseen ids.

**Comment by johntyree** [comment]
Turns out this was a bug anyway. Bug is fixed and test added to make sure `get()` and `__getitem__` do what is expected of them.

File: simplesat/sat/policy.py [hunk 1]
```diff
@@ -190,7 +190,7 @@ def get_next_package_id(self, assignments, clauses):
                     assignments
                 )
 
-        assert assignments[candidate_id] is None, \
+        assert assignments.value(candidate_id) is None, \
```
Comment (cournape) [inline simplesat/sat/policy.py]: Why don't we simply use the `get` method as in the tests ?

Comment (johntyree) [inline simplesat/sat/policy.py]: Hmmm. Yes you're right. I had the relationship between ids and literals backwards. The keys in `AssignmentSet` are ids and thus always positive. `value` is used for looking up *literals*. Here we're looking up an id and should use `get`.

**Pull request merged**